File: backend/app/models/redline.py
"""Output models for redline generation (Phase 6)."""
from __future__ import annotations

from enum import Enum

from pydantic import BaseModel, Field


class DiffOpType(str, Enum):
    EQUAL = "equal"
    INSERT = "insert"
    DELETE = "delete"


class DiffOp(BaseModel):
    type: DiffOpType
    text: str


class RedlineSuggestion(BaseModel):
    doc_id: str
    clause_id: str
    original_text: str
    suggested_text: str
    rationale: str
    diff: list[DiffOp] = Field(default_factory=list)
    diff_markdown: str

File: frontend/src/types/redline.ts
export type DiffOpType = 'equal' | 'insert' | 'delete'

export interface DiffOp {
  type: DiffOpType
  text: string
}

export interface RedlineSuggestion {
  doc_id: string
  clause_id: string
  original_text: string
  suggested_text: string
  rationale: string
  diff: DiffOp[]
  diff_markdown: string
}
